File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/groups/GroupsCreateChildRequest.java
package com.github.seratch.jslack.api.methods.request.groups;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class GroupsCreateChildRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `groups:write`
     */
    private String token;

    /**
     * Private channel to clone and archive.
     */
    private String channel;

    GroupsCreateChildRequest(String token, String channel) {
        this.token = token;
        this.channel = channel;
    }

    public static GroupsCreateChildRequestBuilder builder() {
        return new GroupsCreateChildRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getChannel() {
        return this.channel;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public static class GroupsCreateChildRequestBuilder {
        private String token;
        private String channel;

        GroupsCreateChildRequestBuilder() {
        }

        public GroupsCreateChildRequest.GroupsCreateChildRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public GroupsCreateChildRequest.GroupsCreateChildRequestBuilder channel(String channel) {
            this.channel = channel;
            return this;
        }

        public GroupsCreateChildRequest build() {
            return new GroupsCreateChildRequest(token, channel);
        }

        public String toString() {
            return "GroupsCreateChildRequest.GroupsCreateChildRequestBuilder(token=" + this.token + ", channel=" + this.channel + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/groups/GroupsCreateChildRequest.kt
package com.github.seratch.jslack.api.methods.request.groups

import work.socialhub.kslack.api.methods.SlackApiRequest

class GroupsCreateChildRequest internal constructor(
    /**
     * Authentication token. Requires scope: `groups:write`
     */
    override var token: String?,
    /**
     * Private channel to clone and archive.
     */
    var channel: String?
) : SlackApiRequest {
    class GroupsCreateChildRequestBuilder internal constructor() {
        private var token: String? = null
        private var channel: String? = null

        fun token(token: String?): GroupsCreateChildRequestBuilder {
            this.token = token
            return this
        }

        fun channel(channel: String?): GroupsCreateChildRequestBuilder {
            this.channel = channel
            return this
        }

        fun build(): GroupsCreateChildRequest {
            return GroupsCreateChildRequest(token, channel)
        }

        override fun toString(): String {
            return "GroupsCreateChildRequest.GroupsCreateChildRequestBuilder(token=" + this.token + ", channel=" + this.channel + ")"
        }
    }

    companion object {
        fun builder(): GroupsCreateChildRequestBuilder {
            return GroupsCreateChildRequestBuilder()
        }
    }
}